### PR TITLE
Fix track_running_stats in batchnorm

### PR DESCRIPTION
**Pull request by madt2709**
-`test_batchnorm2d_no_running_stats` and `test_batchnorm3d_no_running_stats` catch the regression. Added two other tests for completeness.

### CODE REVIEW

**Comment by madt2709** [comment]
Closing while I look into failing test

File: test/test_schedule.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def test_fold_conv_batchnorm_notrain(self):
     with Tensor.train(False):
       img = Tensor.empty(1,3,8,8)
       c1 = nn.Conv2d(3,32,3)
-      bn = nn.BatchNorm2d(32, track_running_stats=False)
+      bn = nn.BatchNorm2d(32, track_running_stats=True)
```
Comment (geohot) [inline test/test_schedule.py:196]: Can you add the other test too?

**Comment by geohot** [comment]
Checklist:
- [ ] Do you completely understand what you are doing / changing?
- [ ] Have you put in the effort to make this PR as good as you possibly can?
- [ ] Is this PR to the point where if you were a maintainer of tinygrad that you'd just click merge?

File: test/test_schedule.py (hunk 1 above)
Comment (madt2709) [reply test/test_schedule.py:196]: Added.

**Comment by madt2709** [comment]
@geohot IMO I've ticked off the items on your checklist.
